### PR TITLE
Add functions and startup commands

### PR DESCRIPTION
**Pull request by Earthcomputer**
Adds two commands: `/cfunction` and `/cstartup`.

A specially named function (based on the server ip / world name) is automatically run on "startup" (server/world join) if present, which by default calls a common startup function. The `/cstartup` command is just a convenience for creating and editing these special functions.

### CODE REVIEW

File: src/main/java/net/earthcomputer/clientcommands/features/ClientCommandFunctions.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.IntConsumer;
+import java.util.stream.Stream;
+
+public class ClientCommandFunctions {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    private static final Path FUNCTION_DIR = FabricLoader.getInstance().getConfigDir().resolve("clientcommands").resolve("functions");
```
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/features/ClientCommandFunctions.java]: I suppose this applies to all commands involving file storage, but should we use the static variable `configDir` from `ClientCommands.java`? If not we could make that variable a local variable.

File: src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                .then(literal("edit")
+                    .executes(ctx -> editStartupCommand(true)))));
+    }
+
+    private static int addStartupCommand(FabricClientCommandSource source, boolean global, String command) throws CommandSyntaxException {
+        ensureStartupFileExists(global);
+
+        Path file = global ? ClientCommandFunctions.getGlobalStartupFunction() : ClientCommandFunctions.getLocalStartupFunction();
+        if (file == null) {
+            throw COULD_NOT_CREATE_FILE_EXCEPTION.create();
```
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java:49]: No file is being created here, right?

File: src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int editStartupCommand(boolean global) throws CommandSyntaxException {
+        ensureStartupFileExists(global);
+
+        Path file = global ? ClientCommandFunctions.getGlobalStartupFunction() : ClientCommandFunctions.getLocalStartupFunction();
+        if (file == null) {
+            throw COULD_NOT_CREATE_FILE_EXCEPTION.create();
```
Comment (xpple) [inline src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java:69]: Same here.

File: src/main/java/net/earthcomputer/clientcommands/features/ClientCommandFunctions.java (hunk 1 above)
Comment (Earthcomputer) [reply src/main/java/net/earthcomputer/clientcommands/features/ClientCommandFunctions.java]: I forgot that existed

File: src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java (hunk 2 above)
Comment (Earthcomputer) [reply src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java:49]: If `file` is null, that means that it was impossible to create the file a few lines above.

**Pull request merged**